### PR TITLE
Update numba from 0.56.4 to 0.59.1

### LINKED ISSUE #2698: Update numba
Numba dependency was introduced to the exact 0.56.4 version in: https://github.com/huggingface/dataset-viewer/pull/1026/files#diff-e127bcf65702533dbd7e316815e4b116552c3adef5d4d50db54147d430d868edR24
- #1026

If there is not a strong requirement to use that specific version, we should update it.

### PR DESCRIPTION
**Pull request by albertvillanova**
Update numba from 0.56.4 to 0.59.1.

Fix #2698.

### CODE REVIEW

**Comment by albertvillanova** [comment]
I have removed the explicit dependence on numba-0.56.4 from libcommon/pyproject.toml, as we do not use it in our code.
- Note the dependence is now indirect via: datasets > librosa > numba

Before merging, I would suggest waiting for @lhoestq reply on: https://github.com/huggingface/dataset-viewer/issues/2698#issuecomment-2049108864

**Comment by albertvillanova** [comment]
It seems this PR could be merged without any further impact, @huggingface/dataset-viewer.

**Review by severo — APPROVED** [review]

**Pull request merged**
